### PR TITLE
Project setup

### LINKED ISSUE #1: Initial project setup
**Description**

pom.xml with needed plugins, README, .gitignore, .github workflow files.

### PR DESCRIPTION
**Pull request by 51-code**
Fixes #1 .

Adds all necessary project and workflow files, pom.xml with required plugins and an initial version of README.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -3,33 +3,36 @@
 // 2. Update the issues link in Contributing section in the readme file
 // 3. Update the discussion link in config.yml file in .github/ISSUE_TEMPLATE directory
 
-= repo-template
+= Teragrep Java Configuration Wrapper for Typesafe Config
 
 // Add a short description of your project. Tell what your project does and what it's used for.
 
-This is a template repository for Teragrep organization.
+CNF-02 is a library that provides immutable configuration for Java projects mainly for the Typesafe Config. Typesafe Config will be converted to https://github.com/teragrep/cnf_01[CNF-01]'s Configuration interface, so it is wise to read through its' documentation as well. It has CNF-01 as a dependency, so it provides all the functionality that it has.
```
Comment (ronja-ui) [inline README.adoc]: "...so it is wise to read through *its'* documentation as well."

remove ' and use just *its*. It-pronominin omistusmuoto on aina its eikä siihen tarvita muita lisukkeita merkkaamaan sitä.

File: README.adoc [hunk 2]
```diff
@@ -3,33 +3,36 @@
...
-See the official documentation on https://docs.teragrep.com[docs.teragrep.com].
+- Converts Typesafe Config to CNF-01 interface.
+- Provides all functionalities from CNF-01 with a transitive dependency to the component.
 
 == Limitations
 
-// If your project has limitations, please list them. Otherwise remove this section.
+Features are still under development.
 
 == How to [compile/use/implement]
```
Comment (ronja-ui) [inline README.adoc]: Use either:

- How to compile
- How to use
- How to implement

File: README.adoc (hunk 1 above)
Comment (51-code) [reply README.adoc]: Fixed

File: README.adoc (hunk 2 above)
Comment (51-code) [reply README.adoc]: Fixed

**Review by ronja-ui — APPROVED** [review]

File: README.adoc [hunk 3]
```diff
@@ -3,33 +3,36 @@
 // 2. Update the issues link in Contributing section in the readme file
 // 3. Update the discussion link in config.yml file in .github/ISSUE_TEMPLATE directory
 
-= repo-template
+= Teragrep Java Configuration Wrapper for Typesafe Config
```
Comment (kortemik) [inline README.adoc]: remove Java from the name

File: README.adoc [hunk 4]
```diff
@@ -3,33 +3,36 @@
...
 
 // Add a short description of your project. Tell what your project does and what it's used for.
 
-This is a template repository for Teragrep organization.
+CNF-02 is a library that provides immutable configuration for Java projects mainly for the Typesafe Config. Typesafe Config will be converted to https://github.com/teragrep/cnf_01[CNF-01]'s Configuration interface, so it is wise to read through its documentation as well. It has CNF-01 as a dependency, so it provides all the functionality that it has.
```
Comment (kortemik) [inline README.adoc]: for projects using Java

File: license-header [hunk 5]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Teragrep Java Configuration Wrapper for Typesafe Config (cnf_02)
```
Comment (kortemik) [inline license-header]: remove Java from the name

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,603 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.teragrep</groupId>
+  <artifactId>cnf_02</artifactId>
+  <version>${revision}${sha1}${changelist}</version>
+  <packaging>jar</packaging>
+  <name>cnf_02</name>
+  <description>Teragrep Java Configuration Wrapper for Typesafe Config</description>
```
Comment (kortemik) [inline pom.xml]: remove Java from the name

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,603 @@
...
+            </family>
+          </families>
+          <licenses>
+            <license>
+              <all>
+                <copyright>
+                  <owner>Suomen Kanuuna Oy</owner>
+                  <start>2024</start>
+                </copyright>
+                <text>Teragrep Java Configuration Wrapper for Typesafe Config (cnf_02)</text>
```
Comment (kortemik) [inline pom.xml]: remove Java from the name

**Review by kortemik — CHANGES_REQUESTED** [review]
@ronja-ui see these and request appropriate documentation updates

**Review by kortemik — APPROVED** [review]

**Pull request merged**